### PR TITLE
#4762 Provide information about colorPicker Ui Component

### PR DESCRIPTION
**Pull request by diazwatson**
## Purpose of this pull request

This pull request provides information for the colorPicker Ui Component #4762

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.3/ui_comp_guide/bk-ui_comps.html
- https://devdocs.magento.com/guides/v2.3/ui_comp_guide/components/ui-colorpicker.html

## Links to Magento source code

- https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Ui/view/base/ui_component/etc/definition/colorPicker.xsd
- https://github.com/magento/magento2/blob/f94b4a805bb08a480ff81d53dbe96fa44e7c4328/app/code/Magento/Ui/view/base/ui_component/etc/definition.map.xml#L102
- https://github.com/magento/magento2/blob/f94b4a805bb08a480ff81d53dbe96fa44e7c4328/app/code/Magento/Ui/view/base/ui_component/etc/definition.xml#L152

whatsnew
Added information and example about how to use the [ColorPicker component](https://devdocs.magento.com/guides/v2.3/ui_comp_guide/components/ui-colorpicker.html).

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+group: ui-components-guide
+title: ColorPicker component
+---
+
+The ColorPicker component uses the [Spectrum](https://bgrins.github.io/spectrum/) and [tinycolor.js](https://bgrins.github.io/TinyColor/) libraries to make it easier to choose and implement color values.
+Must be a child of the [Listing]({{ page.baseurl }}/ui_comp_guide/components/ui-listing-grid.html) or [Form]({{ page.baseurl }}/ui_comp_guide/components/ui-form.html) components.
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
The ColorPicker component must be a child of the [Listing]({{ page.baseurl }}/ui_comp_guide/components/ui-listing-grid.html) or [Form]({{ page.baseurl }}/ui_comp_guide/components/ui-form.html) components.
```

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    <td><code>elementTmpl</code></td>
+    <td>The path to the <code>.html</code> template of the particular field type (color-picker).</td>
+    <td><code>String</code></td>
+    <td><code>ui/form/element/email</code></td>
+  </tr>
+</table>
+
+## Sources files
+
+- [app/code/Magento/Ui/view/base/ui_component/etc/definition/colorPicker.xsd](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Ui/view/base/ui_component/etc/definition/colorPicker.xsd)
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
- [app/code/Magento/Ui/view/base/ui_component/etc/definition/colorPicker.xsd]({{ site.mage2bloburl }}/{{page.guide_version}}/app/code/Magento/Ui/view/base/ui_component/etc/definition/colorPicker.xsd)
```

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+- [app/code/Magento/Ui/view/base/web/js/form/element/colorPicker.js](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Ui/view/base/web/js/form/element/color-picker.js)
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
- [app/code/Magento/Ui/view/base/web/js/form/element/colorPicker.js]({{ site.mage2bloburl }}/{{page.guide_version}}/app/code/Magento/Ui/view/base/web/js/form/element/color-picker.js)
```

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+- [app/code/Magento/Ui/view/base/web/js/lib/knockout/bindings/colorPicker.js](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Ui/view/base/web/js/lib/knockout/bindings/color-picker.js)
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
- [app/code/Magento/Ui/view/base/web/js/lib/knockout/bindings/colorPicker.js]({{ site.mage2bloburl }}/{{page.guide_version}}/app/code/Magento/Ui/view/base/web/js/lib/knockout/bindings/color-picker.js)
```

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+- [app/code/Magento/Ui/view/base/web/templates/form/element/colorPicker.html](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Ui/view/base/web/templates/form/element/color-picker.html)
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
- [app/code/Magento/Ui/view/base/web/templates/form/element/colorPicker.html]({{ site.mage2bloburl }}/{{page.guide_version}}/app/code/Magento/Ui/view/base/web/templates/form/element/color-picker.html)
```

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+- [lib/web/jquery/spectrum/spectrum.js](https://github.com/magento/magento2/blob/2.3-develop/lib/web/jquery/spectrum/spectrum.js)
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
- [lib/web/jquery/spectrum/spectrum.js]({{ site.mage2bloburl }}/{{page.guide_version}}/lib/web/jquery/spectrum/spectrum.js)
```

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+- [lib/web/jquery/spectrum/tinycolor.js](https://github.com/magento/magento2/blob/2.3-develop/lib/web/jquery/spectrum/tinycolor.js)
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
- [lib/web/jquery/spectrum/tinycolor.js]({{ site.mage2bloburl }}/{{page.guide_version}}/lib/web/jquery/spectrum/tinycolor.js)
```

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+<table>
+  <tr>
+    <th>Option</th>
+    <th>Description</th>
+    <th>Type</th>
+    <th>Default Value</th>
+  </tr>
+  <tr>
+    <td><code>colorFormat</code></td>
+    <td>Defines the color format that is displayed in selection tool as well as in input field. Valid formats: <code>hex</code>, <code>rgb</code>, <code>hsl</code>, <code>hsv</code>, <code>name</code>, <code>none</code></td>
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
    <td>Defines the color format displayed in the selection tool and input field. Valid formats: <code>hex</code>, <code>rgb</code>, <code>hsl</code>, <code>hsv</code>, <code>name</code>, <code>none</code></td>
```

File: guides/v2.3/ui_comp_guide/components/ui-colorpicker.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    <td><code>string</code></td>
+    <td>-</td>
+  </tr>
+  <tr>
+    <td><code>colorPickerMode</code></td>
+    <td>Defines the mode that affects available color picker functionality. Valid modes: <code>simple</code>, <code>full</code>, <code>noalpha</code>, <code>palette</code></td>
```
Comment (erikmarr) [inline guides/v2.3/ui_comp_guide/components/ui-colorpicker.md]:
```suggestion
    <td>Defines the mode that affects the available color picker functionality. Valid modes: <code>simple</code>, <code>full</code>, <code>noalpha</code>, <code>palette</code></td>
```

**Comment by erikmarr** [comment]
@diazwatson please update the "Links to Magento source code" section in the PR description so you're linking to the master branch instead of dev branch.

**Comment by diazwatson** [comment]
Hi @erikmarr thanks for reviewing this PR.
Requested changes are done.

**Comment by erikmarr** [comment]
running tests

**Comment by erikmarr** [comment]
running tests

**Comment by erikmarr** [comment]
Thanks @diazwatson!

**Comment by diazwatson** [comment]
Thank you guys for the support @erikmarr @osrecio 👍

**Pull request merged**